### PR TITLE
added a new ClusterHealthCheck

### PR DESCRIPTION
**Pull request by ilkinulas**
* searches  a text  in container logs.

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/docker-compose-rule](https://github.com/palantir/docker-compose-rule), @ilkinulas! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

**Comment by ilkinulas** [comment]
Anyone interested in reviewing this pull request?

File: docker-compose-rule-core/src/main/java/com/palantir/docker/compose/connection/Container.java [hunk 1]
```diff
@@ -149,14 +149,25 @@ public String toString() {
     }
 
     public SuccessOrFailure areAllPortsOpen() {
-        List<Integer> unavaliablePorts = portMappings.get().stream()
+        List<Integer> unavailablePorts = portMappings.get().stream()
```
Comment (fryz) [inline docker-compose-rule-core/src/main/java/com/palantir/docker/compose/connection/Container.java]: Thanks for fixing the typo :)

File: docker-compose-rule-junit4/src/main/java/com/palantir/docker/compose/DockerComposeRule.java [hunk 2]
```diff
@@ -237,6 +237,19 @@ public Builder waitingForHostNetworkedPort(int port, HealthCheck<DockerPort> hea
             return addClusterWait(new ClusterWait(clusterHealthCheck, timeout));
         }
 
+        public Builder waitingForTextAppearsInLogs(String serviceName, String regex) {
```
Comment (fryz) [inline docker-compose-rule-junit4/src/main/java/com/palantir/docker/compose/DockerComposeRule.java]: nit: rename `regex` to `text` so as not to confuse this with an actual regex.

**Comment by rndbblnn** [comment]
Can this be merged soon? Thanks!

**Pull request closed without merging**